persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.RDF;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import edu.unc.lib.dl.fcrepo4.BinaryObject;
+import edu.unc.lib.dl.fcrepo4.ContentContainerObject;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FedoraTransaction;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.TransactionManager;
+import edu.unc.lib.dl.fedora.FedoraException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.metrics.TimerFactory;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.search.solr.service.ObjectPathFactory;
+import edu.unc.lib.dl.util.TombstonePropertySelector;
+import io.dropwizard.metrics5.Timer;
+
+/**
+ * A job for removing objects from the repository and replacing them with tombstones
+ *
+ * @author harring
+ *
+ */
+public class DestroyObjectsJob implements Runnable {
+    private static final Timer timer = TimerFactory.createTimerForClass(DestroyObjectsJob.class);
+
+    private List<PID> objsToDestroy;
+    @Autowired
+    private RepositoryObjectFactory repoObjFactory;
+    @Autowired
+    private RepositoryObjectLoader repoObjLoader;
+    @Autowired
+    private TransactionManager txManager;
+    @Autowired
+    private DestroyProxyService proxyService;
+    @Autowired
+    private ObjectPathFactory pathFactory;
+
+    public DestroyObjectsJob(List<PID> objsToDestroy) {
+        this.objsToDestroy = objsToDestroy;
+    }
+
+    @Override
+    public void run() {
+        FedoraTransaction tx = txManager.startTransaction();
+        try (Timer.Context context = timer.time()) {
+            // convert each destroyed obj to a tombstone
+            for (PID pid : objsToDestroy) {
+                // remove containment relation from obj's parent
+                proxyService.destroyProxy(pid);
+                RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+                // purge tree with repoObj as root from repository
+                destroyTree(repoObj);
+           }
+        } catch (Exception e) {
+             tx.cancel(e);
+        } finally {
+             tx.close();
+        }
+    }
+
+    private void destroyTree(RepositoryObject rootOfTree) throws FedoraException {
+        if (rootOfTree instanceof ContentContainerObject) {
+            ContentContainerObject container = (ContentContainerObject) rootOfTree;
+            List<ContentObject> members = container.getMembers();
+            for (ContentObject member : members) {
+                destroyTree(member);
```

Comment:
I think after you destroy each member you will need to destroy the proxy from the parent to it as well

Reply:
but i actually do that up in the run() method before i even call destroyTree(). do you think that needs to change?

Reply:
that will only remove the first proxy though, from the start of the tree to its parent, right? There will be proxies between all the children and their children if you are destroying a folder that contains folders.

But it might not actually be necessary, its possible fedora will delete the proxies since they are ldp:contained by the thing being deleted. If you can verify that the proxies don't outlive their parents (and leave tombstones all over the place) then what you have here should be good.